sdk/messaging/azservicebus/internal/idle_check.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+type LocalIdleTracker struct {
+	// MaxDuration controls how long we'll wait, on the client side, for the first message
+	MaxDuration time.Duration
+
+	// IdleStart marks the first time the user cancelled in a string of cancellations.
+	// It gets reset any time there is a success or a non-cancel related failure.
+	// NOTE: this is public for some unit tests but isn't intended to be set by external code.
```

Comment:
Maybe instead of exposing the field, you add a `Testing_IdleStart()` method or something like that?

Reply:
I thought about it, I actually set and read it at this point for tests so it'd be me adding in getters and setters, which is probably worse.

Reply:
Ah ok yeah that would be ugly then.